lib/child_process.js

```diff
@@ -19,15 +19,20 @@ const ChildProcess = exports.ChildProcess = child_process.ChildProcess;
 exports.fork = function(modulePath /*, args, options*/) {
 
   // Get options and args arguments.
-  var options, args, execArgv;
-  if (Array.isArray(arguments[1])) {
-    args = arguments[1];
-    options = util._extend({}, arguments[2]);
-  } else if (arguments[1] && typeof arguments[1] !== 'object') {
-    throw new TypeError('Incorrect value of args option');
-  } else {
-    args = [];
-    options = util._extend({}, arguments[1]);
+  var execArgv;
+  var options = {};
+  var args = [];
+  var pos = 1;
+  if (Array.isArray(arguments[pos])) {
```

Comment:
Can you check that `pos < arguments.length`?  I believe V8 still deoptimizes on out-of-bounds arguments access.